src/NexusMods.Networking.NexusWebApi/NexusModsLibrary.Collections.cs

```diff
@@ -262,28 +264,22 @@ static UpdatePolicy ConvertUpdatePolicy(Abstractions.Collections.Json.UpdatePoli
 
         return revisionMetadata.Rebase(_connection.Db);
     }
-    
+
     /// <summary>
-    /// Gets all revision numbers that are newer than the current revision.
+    /// Gets the last published revision number.
     /// </summary>
-    public async ValueTask<RevisionNumber[]> GetNewerRevisionNumbers(
-        CollectionRevisionMetadata.ReadOnly currentRevision,
+    public async ValueTask<Optional<RevisionNumber>> GetLastPublishedRevisionNumber(
+        CollectionMetadata.ReadOnly collection,
         CancellationToken cancellationToken)
     {
-        var revisionNumbers = await GetAllRevisionNumbers(currentRevision.Collection, cancellationToken);
-
-        var currentValue = currentRevision.RevisionNumber.Value;
-        var newer = revisionNumbers
-            .TakeWhile(other => other.Value > currentValue)
-            .ToArray();
-
-        return newer;
+        var revisionNumbers = await GetAllRevisionNumbers(collection, cancellationToken);
+        return revisionNumbers.FirstOrOptional(static tuple => tuple.Status == RevisionStatus.Published).Convert(static tuple => tuple.Number);
     }
 
     /// <summary>
     /// Gets all revision numbers for the given collection in descending order.
     /// </summary>
-    public async ValueTask<RevisionNumber[]> GetAllRevisionNumbers(
+    public async ValueTask<(RevisionNumber Number, RevisionStatus Status)[]> GetAllRevisionNumbers(
```

Comment:
Users can delete collections from Nexus if they only uploaded a draft and never published the first revision.

This means that this query can fail.

Reply:
I'll deal with API errors in a separate PR.